airbyte-webapp/src/hooks/services/useBuildUpdateCheck.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect } from "react";
+import { useIntl } from "react-intl";
+import { fromEvent, interval, merge, throttleTime } from "rxjs";
+
+import { useAppMonitoringService } from "./AppMonitoringService";
+import { useNotificationService } from "./Notification";
+
+interface BuildInfo {
+  build: string;
+}
+
+const INTERVAL = 60_000;
+
+/**
+ * This hook sets up a check to /buildInfo.json, which is generated by the build system on every build
+ * with a new hash. If it ever detects a new hash in it, we know that the Airbyte instance got updated
+ * and show a notification to the user to reload the page.
+ */
+export const useBuildUpdateCheck = () => {
+  const { formatMessage } = useIntl();
+  const { registerNotification } = useNotificationService();
+  const { trackError } = useAppMonitoringService();
+
+  useEffect(() => {
+    // Trigger the check every ${INTERVAL} milliseconds or whenever the window regains focus again
+    const subscription = merge(interval(INTERVAL), fromEvent(window, "focus"))
+      // Throttle it to maximum once every 10 seconds
+      .pipe(throttleTime(10 * 1000))
```

Comment:
Nice clean way to avoid bunching up requests if the focus event comes right before the next interval; I briefly wondered if some `switchMap` shenanigans would be needed to avoid that scenario (and whether it would actually be a problem, given our current scale and the CDN deployment discussions), but this is much better.

nit: I do think that it would be better to be consistent with how the time windows are defined; so `10 * 1000` should become `10_000` and either both or neither number should be extracted to a named binding (no strong preference which):
```ts
// either this...
const INTERVAL = 60_000;
const MINIMUM_WAIT_BEFORE_REFETCH = 10_000;

merge(interval(INTERVAL), fromEvent(window, "focus"))
  .pipe(throttleTime(MINIMUM_WAIT_BEFORE_REFETCH))

// ...or this
merge(interval(60_000), fromEvent(window, "focus"))
  .pipe(throttleTime(10_000))
```